core/src/main/resources/error/error-classes.json

```diff
@@ -1592,7 +1592,7 @@
       },
       "MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY" : {
         "message" : [
-          "Correlated scalar subqueries in the GROUP BY clause must also be in the aggregate expressions<treeNode>"
+          "Correlated scalar subqueries must be aggregated to return at most one row."
```

Comment:
not related to this PR, but how is this different from `MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY_OUTPUT`?

Reply:
Seems like we can integrate it into `MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY`.

Let me address it as follow-up soon.

Reply:
SGTM. Is the ticket created?

Reply:
Here is the ticket just created: https://issues.apache.org/jira/browse/SPARK-42239

Let me address this soon.

Reply:
Just opened the PR: https://github.com/apache/spark/pull/39806